PathPlanning/PotentialFieldPlanning/potential_field_planning.py

```diff
@@ -128,6 +131,19 @@ def potential_field_planning(sx, sy, gx, gy, ox, oy, reso, rr):
         rx.append(xp)
         ry.append(yp)
 
+        if ((None, None) not in previous_id and
+                (previous_id[0] == previous_id[1]
+                 or previous_id[1] == previous_id[2]
+                 or previous_id[0] == previous_id[2])):
+            print("Oscillation detected!!!")
+            print(previous_id)
+            break
+
+        # roll previous
+        previous_id[0] = previous_id[1]
```

Comment:
This code can be used for only when horizon is 3. So, I think "deque" should be used here. https://docs.python.org/3/library/collections.html#collections.deque

Reply:
done

Reply:
done